_data-prepper/common-use-cases/log-enrichment.md

```diff
@@ -0,0 +1,398 @@
+
+In this example, the first processor sets the value of the `debug` key to `true` if the key already exists in the event. The second processor only sets the `debug` key to `true` if the key doesn't exist in the event, because `overwrite_if_key_exists` is set to `true`.
+
+```json
+...
+processor:
+  - add_entries:
+    entries:
+    - key: "debug"
+      value: true 
+...
+processor:
+  - add_entries:
+    entries:
+    - key: "debug"
+      value: true 
+      overwrite_if_key_exists: true
+...
+```
+{% include copy-curl.html %}
+
+You can also use a format string to construct new entries from existing entries. For example, `${date}-${time}` will create a new entry based on the values of the existing entries `date` and `time`.
+
+For example, the following pipeline adds new event entries dynamically from existing events:
+
+```json
+processor:
+  - add_entries:
+    entries:
+    - key: "key_three"
+      format: "${key_one}-${key_two}
+```
+{% include copy-curl.html %}
+
+For example, consider the following incoming event:
+
+```json
+{
+   "key_one": "value_one",
+   "key_two": "value_two"
+}
+```
+{% include copy-curl.html %}
+
+The processor transforms it into an event with a new key `key_three`, which combines values of other keys in the original event, as shown in the following example:
+
+```json
+{
+   "key_one": "value_one",
+   "key_two": "value_two",
+   "key_three": "value_one-value_two"
+}
+```
+{% include copy-curl.html %}
+
+## Mutating strings
+
+The various [Mutate string]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/mutate-string/) processors offer tools to manipulate strings in incoming data. For example, if you need to split a string into an array, use the `split_string` processor:
+
+```json
+...
+processor:
+  - split_string:
+    entries:
+    - source: "message"
+      delimiter: "&"
+...
+```
+{% include copy-curl.html %}
+
+The processor will transform a string such as `a&b&c` into `["a", "b", "c"]`.
+
+## Converting lists to maps
+
+The [List-to-map]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/list-to-map/) processor, which is one of the Mutate events processors, converts a list of objects in an event to a map.
+
+For example, consider the following processor configuration:
+
+```json
+...
+processor:
+    - list_to_map:
+        key: "name"
+        source: "A-car-as-list"
+        target: "A-car-as-map"
+        value_key: "value"
+        flatten: true
+...
+```
+{% include copy-curl.html %}
+
+This processor will convert an event that contains a list of objects like this:
+
+```json
+{
+  "A-car-as-list": [
+    {
+      "name": "make",
+      "value": "tesla"
+    },
+    {
+      "name": "model",
+      "value": "model 3"
+    },
+    {
+      "name": "color",
+      "value": "white"
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+into a map:
```

Comment:
I'm not a huge fan of splitting sentences over multiple lines, but I'll defer to you if this is best practice.